app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,267 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.GestureDetector
+import android.view.GestureDetector.SimpleOnGestureListener
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.widget.HorizontalScrollView
+
+import androidx.annotation.IntRange
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.children
+import androidx.core.view.forEach
+import androidx.core.view.updateLayoutParams
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging.Companion.get
+import nerd.tuxmobil.fahrplan.congress.R
+import nerd.tuxmobil.fahrplan.congress.schedule.HorizontalSnapScrollState.Companion.FLING_COLUMN_MULTIPLIER
+import nerd.tuxmobil.fahrplan.congress.schedule.HorizontalSnapScrollState.Companion.SWIPE_THRESHOLD_VELOCITY
+import kotlin.math.ceil
+import kotlin.math.floor
+import kotlin.math.roundToInt
+
+import androidx.core.view.get
+
+class HorizontalSnapScrollView(context: Context?, attrs: AttributeSet?) :
+    HorizontalScrollView(context, attrs) {
+    private val logging = get()
+    private val gestureDetector: GestureDetector
+    private var horizontalSnapScrollState = HorizontalSnapScrollState(logging)
+    private lateinit var roomNames: HorizontalScrollView
+
+    /**
+     * get currently displayed column index
+     *
+     * @return index (0..n)
+     */
+    val columnIndex: Int
+        get() = horizontalSnapScrollState.activeColumnIndex
+
+    internal inner class YScrollDetector : SimpleOnGestureListener() {
+        override fun onDown(motionEvent: MotionEvent): Boolean {
+            val xStart = motionEvent.x.toInt()
+//            logging.d(LOG_TAG, "onDown xStart: " + xStart + " getMeasuredWidth: " + getMeasuredWidth())
+            val ofs = (scrollX * horizontalSnapScrollState.displayColumnCount).toFloat() / measuredWidth
+            horizontalSnapScrollState = horizontalSnapScrollState.copy(
+                logging = horizontalSnapScrollState.logging,
+                xStart = xStart,
+                displayColumnCount = horizontalSnapScrollState.displayColumnCount,
+                roomsCount = horizontalSnapScrollState.roomsCount,
+                columnWidth = horizontalSnapScrollState.columnWidth,
+                activeColumnIndex = ofs.roundToInt()
+            )
+            return super.onDown(motionEvent)
+        }
+
+        override fun onFling(
+            start: MotionEvent,
+            end: MotionEvent,
+            velocityX: Float,
+            velocityY: Float
+        ): Boolean {
+            val normalizedVelocityX = velocityX / resources.displayMetrics.density
+            val columns = ceil((normalizedVelocityX / SWIPE_THRESHOLD_VELOCITY * FLING_COLUMN_MULTIPLIER).toDouble()).toInt()
```

Comment:
`FLING_COLUMN_MULTIPLIER` can just be a constant in the `HorizontalSnapScrollView` class. It does not belong to the `HorizontalSnapScrollViewState` class.

Reply:
Looks like `FLING_COLUMN_MULTIPLIER` can be `private`.